File: 정수 제곱근 판별.py
"""
def solution(n):

    for i in range(10000001):
        if i*i == n:
            return (i+1)*(i+1)

    return -1
"""
#2022.11.12
def solution(n):
    if n ** (1/2) == int(n ** (1/2)):
        return (n**(1/2) +1)**2
    return -1

File: 정수 제곱근 판별.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <math.h>

long long solution(long long n) {
    long long answer = 0;
    double a = pow((double) n, (double) 0.5);
    if (a == (int)a){
        a++;
        return a*a;
    }

    return -1;
}
